src/app/modules/item/pages/group-progress-grid/group-progress-grid.component.html

```diff
@@ -3,104 +3,120 @@
   (change)="onFilterChange($event)"
 ></alg-composition-filter>
 
-<ng-container *ngIf="state$ | async as state">
-  <alg-error
-    *ngIf="state.isError; else noError"
-    class="dark"
-    i18n-message message="Error while loading the data"
-    icon="fa fa-exclamation-triangle"
-    [showRefreshButton]="true"
-    (refresh)="refresh()"
-  ></alg-error>
+<ng-container *ngIf="columns$ | async as columnsState">
+  <ng-container *ngIf="rows$ | async as rowsState">
+    <alg-error
+      *ngIf="rowsState.isError || columnsState.isError; else noError"
+      class="dark"
+      i18n-message message="Error while loading the data"
+      icon="fa fa-exclamation-triangle"
+      [showRefreshButton]="true"
+      (refresh)="refresh()"
+    ></alg-error>
 
-  <ng-template #noError>
-    <div class="loader" *ngIf="state.isFetching && (!state.data || state.data.rows.length === 0)">
-      <alg-loading></alg-loading>
-    </div>
+    <ng-template #noError>
+      <div class="loader" *ngIf="rowsState.isFetching && (!rowsState.data || rowsState.data.length === 0)">
+        <alg-loading></alg-loading>
```

Comment:
`rowState` depends on `columnsState`, therefore if `columnsState` is loading, `rowsState` is necessarily loading too.
Now for the sake of readability, it might be clearer to add a `... || columnsState.loading` here.
I took the side not to add it because I considered the template are colocated and deeply related, and therefore information is contained in the component part. If you prefer to add it tell me :relaxed: